api/v1beta1/kafkacluster_types.go

```diff
@@ -189,6 +206,11 @@ type Broker struct {
 
 // BrokerConfig defines the broker configuration
 type BrokerConfig struct {
+	// processRoles defines the role(s) for this particular Kafka node: "broker", "controller", or both.
+	// This must be set in KRaft mode. If set in ZooKeeper mode, Koperator ignores this configuration.
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Roles                []string                      `json:"processRoles,omitempty"`
```

Comment:
This also can be []KraftNodeRole

Reply:
Well, I thought about different potential names, but ended up following what the Kafka community itself uses: https://github.com/apache/kafka/blob/trunk/config/kraft/controller.properties#L24

This way, the users (I assume they are someone familiar with Kafka) would immediately know what this field is about. Hope that makes sense